### PR TITLE
WG LC preparation

### LINKED ISSUE #40: Add an example for a path computation request
In a later version of the ID, it may make sense to include an actual example for a path computation request (e.g., for XML encoding in NETCONF).

See comment 15. in issue #37 

- [x] Waiting for a stable YANG model

### LINKED ISSUE #89: Move groupings to ietf-te or ietf-te-types
The following two groupings have been defined by copying a sub-set of the attributes defined in ietf-te YANG model:

https://github.com/rvilalta/ietf-te-path-computation/blob/826da8036051a1eb25698f4ba735dc4c8b591b4a/ietf-te-path-computation.yang#L508-L562

It is worthwhile checking if these two groupings can be moved to either ietf-te or ietf-te-types and re-used by both ietf-te and ietf-te-path-computation YANG models

- [x] Pending resolution of https://github.com/tsaad-dev/te/issues/150

### LINKED ISSUE #92: What about Appendix A?
Appendix A provides information which has been quite valuable during TEAS WG discussion to justify the need for a Path Computation RPC

However, the text in the appendix is not ready to be published in an RFC

It is not clear whether keeping appendix A in the RFC is worth the effort of updating the text or whether removing the appendix would simplify moving the document forward

- [x] Notify TEAS WG about the proposal to remove appendix A
- [x] Remove appendix A

### LINKED ISSUE #98: Align with latest update of draft-ietf-teas-yang-te-28
Align the draft with latest update of draft-ietf-teas-yang-te (new versions after draft-ietf-teas-yang-te-28)

In particular, consider the resolution of:
- https://github.com/tsaad-dev/te/issues/174, which impact the JSON examples
- https://github.com/tsaad-dev/te/issues/175, which impact the security consideration sections

### PR DESCRIPTION
**Pull request by italobusi**
I-D text updates:
- [x] Updated to address Tom's comments in #95 
- [x] Removed appendix A (fix #92 )
- [x] Add example (fix #40 )
- [x] Align securing considerations with https://datatracker.ietf.org/doc/html/[draft-ietf-teas-yang-te-29](https://datatracker.ietf.org/doc/html/draft-ietf-teas-yang-te-29) (fix #98 )

YANG updates:

- [x] Deleted the encoding-and-switching-type and the tunnel-common-attributes groupings and used the new groupings from ietf-te (Fix #89 )

### CODE REVIEW

File: draft-ietf-teas-yang-path-computation.txt [hunk 1]
```diff
@@ -0,0 +1,4704 @@
...
+                                                         24 January 2022
+
+
+           A YANG Data Model for requesting path computation
+                draft-ietf-teas-yang-path-computation-17
+
+Abstract
+
+   There are scenarios, typically in a hierarchical Software-Defined
+   Networking (SDN) context, where the topology information provided by
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt]:
```suggestion
   Networking (SDN) architecture, where the topology information provided by
```

File: draft-ietf-teas-yang-path-computation.txt [hunk 2]
```diff
@@ -0,0 +1,4704 @@
...
+   a Traffic Engineering (TE) network provider may be insufficient for
+   its client to perform end-to-end path computation.  In these cases
+   the client would need to request the provider to calculate some
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt]: This sentence is not clear.

File: draft-ietf-teas-yang-path-computation.txt [hunk 3]
```diff
@@ -0,0 +1,4704 @@
...
+   (partial) feasible paths.
+
+   This document defines a YANG data model for a Remote Procedure Call
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt:32]: Does this document really define a data model or does it define RPCs using the YANG modeling language?

File: draft-ietf-teas-yang-path-computation.txt [hunk 4]
```diff
@@ -0,0 +1,4704 @@
...
+   (RPC) to request path computation.  This model complements the
+   solution, defined in RFC YYYY, to configure a TE tunnel path in
+   "compute-only" mode.
+
+   [RFC EDITOR NOTE: Please replace RFC YYYY with the RFC number of
+   draft-ietf-teas-yang-te once it has been published.
+
+   Moreover this document describes some use cases where a path
+   computation request, via YANG-based protocols (e.g., NETCONF or
+   RESTCONF), can be needed.
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt:42]:
```suggestion
   Moreover, this document describes some use cases where the path
   computation RPCs defined in this document can be used utilizing the
   NETCONF or RESTCONF protocol
```

File: draft-ietf-teas-yang-path-computation.txt [hunk 5]
```diff
@@ -0,0 +1,4704 @@
...
+1.  Introduction
+
+   There are scenarios, typically in a hierarchical Software-Defined
+   Networking (SDN) context, where the topology information provided by
+   a Traffic Engineering (TE) network provider may be insufficient for
+   its client to perform end-to-end path computation.  In these cases
+   the client would need to request the provider to calculate some
+   (partial) feasible paths, complementing its topology knowledge, to
+   make end-to- end path computation feasible.
+
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt:148]: See comments on the Abstract.

File: draft-ietf-teas-yang-path-computation.txt [hunk 6]
```diff
@@ -0,0 +1,4704 @@
...
+   These types of scenarios can be applied to different interfaces in
+   different reference architectures:
+
+   *  Application-Based Network Operations (ABNO) control interface
+      [RFC7491], in which an Application Service Coordinator can request
+      ABNO controller to take in charge path calculation (see Figure 1
+      in [RFC7491]).
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt:155]: Editorial:
```suggestion
     in which an Application Service Coordinator can request the
      ABNO controller to perform path calculation  and return the
      path to the Application Service Coordinator (see Figure 1
      in [RFC7491]).
```

File: draft-ietf-teas-yang-path-computation.txt [hunk 7]
```diff
@@ -0,0 +1,4704 @@
...
+
+   *  Abstraction and Control of TE Networks (ACTN) [RFC8453], where a
```
Comment (dieterbeller) [inline draft-ietf-teas-yang-path-computation.txt:157]: Editorial
```suggestion
   *  Abstraction and Control of TE Networks (ACTN) [RFC8453], where a
      controller hierarchy is defined.
      In the ACTN context, path computation is needed on the interface
      between Customer Network Controller (CNC)  and Multi-Domain
      Service Coordinator (MDSC), called CNC-MDSC Interface (CMI),
      and on the interface between MSDC and Provisioning Network
      Controller (PNC), called MDSC-PNC Interface  (MPI). 
      [RFC8454] describes an information model for the Path
      Computation request.
```

File: draft-ietf-teas-yang-path-computation.txt (hunk 1 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt:26]: Fixed in https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 2 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt]: Proposed rephrase in: https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 3 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt]: RPCs are part of YANG data models. This YANG data model contains only RPCs

Text updated to improve the clarity in: https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 4 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt:42]: Fixed in https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 5 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt:148]: Text updated to improve clarity in: https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 6 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt]: Fixed in https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

File: draft-ietf-teas-yang-path-computation.txt (hunk 7 above)
Comment (italobusi) [reply draft-ietf-teas-yang-path-computation.txt:157]: Fixed in https://github.com/rvilalta/ietf-te-path-computation/pull/96/commits/fc6c5456c7cfaf1ec3ffa97477f75847d2fd428f

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

**Pull request merged**